lntest/itest/assertions.go

```diff
@@ -623,6 +623,121 @@ func txStr(chanPoint *lnrpc.ChannelPoint) string {
 	return cp.String()
 }
 
+// waitForChannelUpdate waits for a node to receive the expected channel
+// updates.
+func waitForChannelUpdate(t *harnessTest, subscription graphSubscription,
+	expUpdates []expectedChanUpdate) {
+
+	// Create an array indicating which expected channel updates we have
+	// received.
+	found := make([]bool, len(expUpdates))
+out:
+	for {
+		select {
+		case graphUpdate := <-subscription.updateChan:
+			for _, update := range graphUpdate.ChannelUpdates {
+				require.NotZerof(
+					t.t, len(expUpdates),
+					"received unexpected channel "+
+						"update from %v for channel %v",
+					update.AdvertisingNode,
+					update.ChanId,
+				)
+
+				// For each expected update, check if it matches
+				// the update we just received.
+				for i, exp := range expUpdates {
+					fundingTxStr := txStr(update.ChanPoint)
+					if fundingTxStr != txStr(exp.chanPoint) {
+						continue
+					}
+
+					if update.AdvertisingNode !=
```

Comment:
Also continue if `found[i]` == true? Otherwise that slice doesn't make any sense.
And since we're only ever calling `waitForChannelUpdate()` with a single update, maybe this "waiting for a slice of updates" is a premature optimization?